### PR TITLE
Respect dynamic logging level changes for LevelSwitch section

### PR DESCRIPTION
**Pull request by skomis-mm**
- dynamic level changes notifications for `LevelSwitch` section
- tests added
- NRE fix for `DummyConsoleSink`
- unused / unreachable related junk removed
- sample updated

Fixes [#99](https://github.com/serilog/serilog-settings-configuration/issues/99)

### CODE REVIEW

**Comment by skomis-mm** [comment]
Need some investigation why test on Travis / Linux fails. Probably should change file based notifications to something more stable

**Comment by skomis-mm** [comment]
Ok, ready for review when someone have a time :)

File: test/Serilog.Settings.Configuration.Tests/DynamicLevelChangeTests.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                logger.ForContext(Constants.SourceContextPropertyName, "Root.Test").Write(Some.DebugEvent());
+                Assert.Single(DummyConsoleSink.Emitted);
+
+                DummyConsoleSink.Emitted.Clear();
+                UpdateConfig(overrideLevel: LogEventLevel.Debug);
+                logger.ForContext(Constants.SourceContextPropertyName, "Root.Test").Write(Some.DebugEvent());
+                Assert.Single(DummyConsoleSink.Emitted);
+            }
+
+            void UpdateConfig(LogEventLevel? minimumLevel = null, LogEventLevel? switchLevel = null, LogEventLevel? overrideLevel = null)
```
Comment (tsimbalar) [inline test/Serilog.Settings.Configuration.Tests/DynamicLevelChangeTests.cs]: Maybe that's personal taste, but I think turning the `UpdateConfig` Local Function into a proper method would make it clearer (and shorten the code in the test method)

Comment (skomis-mm) [reply test/Serilog.Settings.Configuration.Tests/DynamicLevelChangeTests.cs]: Agree, thanks.

**Pull request merged**